src/types/offline-error.ts

```diff
@@ -0,0 +1 @@
+export default class OfflineError extends Error {}
```

Comment:
Why create a brand new type here if its just a copy of `Error`?

Reply:
I did this for type checking. There are cases where I want to handle a timeout differently than just a regular error. 

```
if (err instance of OfflineError) {
  // do something different
}
```